### DISCUSSION TITLE
Passing a small input sequence

### DISCUSSION THREAD

**Discussion by Sinjini15** [archinetai/audio-diffusion-pytorch#12] [General]
For the tutorial on the repo page if I try to pass 

```
model = AudioDiffusionModel(in_channels=1)
x = torch.randn(2, 1, 2 ** 6)
loss = model(x)
```

I get the following error:

`RuntimeError: Sizes of tensors must match except in dimension 1. Expected size 2 but got size 1 for tensor number 1 in the list.`

Seems like for anything less than `2 ** 14` I get this error. Is there any way to pass a smaller input sequence to the diffusion model?

**Comment by flavioschneider** [comment]
You have to reduce the network `patch_size` and downsampling `factors`. Note however that `2**6` is a length of only 64, I doubt you can listen to anything meaningful with so few numbers (even very bad quality music has 6000 to 12000 samples per second). If you are doing something else that is not audio, you might want to do diffusion on something else that is not a `UNet1d` for such a short length. 
```py 
patch_size=16, # This divides the input by 2**4
multipliers=[1, 2, 4, 4, 4, 4, 4], # (len=num_layers+1)
factors=[4, 4, 4, 2, 2, 2], # This divides the input the respective numbers (len=num_layers)
num_blocks=[2, 2, 2, 2, 2, 2], #  (len=num_layers)
attentions=[False, False, False, True, True, True], #  (len=num_layers)
```

**Reply by Sinjini15** [reply]
Thanks for explaining! I am just playing around with some other data, not specifically music :)

My data size is actually `2**9` but I was testing with 64 length data to see if it would even work. Your suggestion worked! 

Posting for other people who might want to implement the same:

I reduced `patch_size` to 8 and `factors` to `[2, 2, 2, 1, 1, 1]` and it worked for me!
